ledger/accountdb.go

```diff
@@ -786,6 +878,57 @@ type resourcesData struct {
 	UpdateRound   uint64        `codec:"z"`
 }
 
+func (rd *resourcesData) IsHolding() bool {
+	return (rd.ResourceFlags & resourceFlagsNotHolding) == resourceFlagsHolding
+}
+
+func (rd *resourcesData) IsOwning() bool {
+	return (rd.ResourceFlags & resourceFlagsOwnership) == resourceFlagsOwnership
+}
+
+func (rd *resourcesData) IsEmptyApp() bool {
+	return rd.SchemaNumUint == 0 &&
+		rd.SchemaNumByteSlice == 0 &&
+		len(rd.KeyValue) == 0 &&
+		len(rd.ApprovalProgram) == 0 &&
+		len(rd.ClearStateProgram) == 0 &&
+		len(rd.GlobalState) == 0 &&
+		rd.LocalStateSchemaNumUint == 0 &&
+		rd.LocalStateSchemaNumByteSlice == 0 &&
+		rd.GlobalStateSchemaNumUint == 0 &&
+		rd.GlobalStateSchemaNumByteSlice == 0 &&
+		rd.ExtraProgramPages == 0
+}
+
+func (rd *resourcesData) IsApp() bool {
+	if (rd.ResourceFlags & resourceFlagsEmptyApp) == resourceFlagsEmptyApp {
+		return true
+	}
+	return !rd.IsEmptyApp()
+}
+
+func (rd *resourcesData) IsEmptyAsset() bool {
+	return rd.Amount == 0 &&
+		!rd.Frozen &&
+		rd.Total == 0 &&
+		rd.Decimals == 0 &&
+		!rd.DefaultFrozen &&
+		rd.UnitName == "" &&
+		rd.AssetName == "" &&
+		rd.URL == "" &&
+		rd.MetadataHash == [32]byte{} ||
+		rd.Manager.IsZero() ||
+		rd.Reserve.IsZero() ||
+		rd.Freeze.IsZero() ||
+		rd.Clawback.IsZero()
```

Comment:
should be `&&`. Otherwise a record with only holding set returns True. I'll fix with a PR with a test